internal/tss/services/pool_populator.go

```diff
@@ -0,0 +1,245 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/wallet-backend/internal/entities"
+	"github.com/stellar/wallet-backend/internal/services"
+	"github.com/stellar/wallet-backend/internal/tss"
+	"github.com/stellar/wallet-backend/internal/tss/router"
+	"github.com/stellar/wallet-backend/internal/tss/store"
+)
+
+type PoolPopulator interface {
+	PopulatePools(ctx context.Context) error
+}
+
+type poolPopulator struct {
+	Router     router.Router
+	Store      store.Store
+	RPCService services.RPCService
+}
+
+func NewPoolPopulator(router router.Router, store store.Store, rpcService services.RPCService) (*poolPopulator, error) {
+	if router == nil {
+		return nil, fmt.Errorf("router is nil")
+	}
+	if store == nil {
+		return nil, fmt.Errorf("store is nil")
+	}
+	if rpcService == nil {
+		return nil, fmt.Errorf("rpcservice is nil")
+	}
+	return &poolPopulator{
+		Router:     router,
+		Store:      store,
+		RPCService: rpcService,
+	}, nil
+}
+
+func (p *poolPopulator) PopulatePools(ctx context.Context) error {
+	err := p.routeNewTransactions(ctx)
+	if err != nil {
+		return fmt.Errorf("error routing new transactions: %w", err)
+	}
+
+	err = p.routeErrorTransactions(ctx)
+	if err != nil {
+		return fmt.Errorf("error routing new transactions: %w", err)
+	}
+
+	err = p.routeFinalTransactions(ctx, tss.RPCTXStatus{RPCStatus: entities.FailedStatus})
+	if err != nil {
+		return fmt.Errorf("error routing failed transactions: %w", err)
+	}
+
+	err = p.routeFinalTransactions(ctx, tss.RPCTXStatus{RPCStatus: entities.SuccessStatus})
+	if err != nil {
+		return fmt.Errorf("error routing successful transactions: %w", err)
+	}
+
+	err = p.routeNotSentTransactions(ctx)
+	if err != nil {
+		return fmt.Errorf("error routing not_sent transactions: %w", err)
+	}
+	return nil
+}
```

Comment:
Quick question. Should one step impact the other? In this approach, if the first one fails, we will never execute the other steps.

Reply:
Good point! Although it doesnt really matter as this runs on a ticker, I changed this to simply log the errors and not return them. So now, even if routeNewTransactions fails, the others can continue